spark/src/main/scala/geotrellis/spark/io/cog/COGValueReader.scala

```diff
@@ -71,13 +72,69 @@ trait COGValueReader[ID] {
       } catch {
         case th: Throwable => exceptionHandler(key)(th)
       }
+
+    }
+
+    def readSubsetBands(key: K, bands: Seq[Int]): Array[Option[Tile]] = {
+      val (zoomRange, spatialKey, overviewIndex, gridBounds) =
+        cogLayerMetadata.getReadDefinition(key.getComponent[SpatialKey], layerId.zoom)
+
+      val baseKeyIndex = keyIndexes(zoomRange)
+
+      val maxWidth = Index.digits(baseKeyIndex.toIndex(baseKeyIndex.keyBounds.maxKey))
+      val uri = fullPath(keyPath(key.setComponent(spatialKey), maxWidth, baseKeyIndex, zoomRange))
+
+      val sourceGeoTiff =
+        GeoTiffReader.readMultiband(uri, streaming = true)
+
+      // Have to convert the sourceGeoTiff to a GeoTiffMultibandTile
+      // in order to access the desired crop method.
+      val sourceMultibandTile =
+        GeoTiffMultibandTile(
```

Comment:
I suppose It's possible to use pattern matching here, but this solution is also good.